qiskit/circuit/quantumcircuit.py

```diff
@@ -2205,27 +2205,41 @@ def measure_active(self, inplace: bool = True) -> Optional["QuantumCircuit"]:
         else:
             return None
 
-    def measure_all(self, inplace: bool = True) -> Optional["QuantumCircuit"]:
+    def measure_all(
+        self, inplace: bool = True, add_creg: bool = True
+    ) -> Optional["QuantumCircuit"]:
         """Adds measurement to all qubits. Creates a new ClassicalRegister with a
-        size equal to the number of qubits being measured.
+        size equal to the number of qubits being measured if `add_creg = True`.
 
         Returns a new circuit with measurements if `inplace=False`.
 
         Args:
             inplace (bool): All measurements inplace or return new circuit.
+            add_creg (bool): Adds a new classical register.
 
         Returns:
             QuantumCircuit: Returns circuit with measurements when `inplace = False`.
+
+        Raises:
+            CircuitError: if `add_creg = False` and `len(circ.clbits) < len(circ.qubits)`.
         """
         if inplace:
             circ = self
         else:
             circ = self.copy()
-
-        new_creg = circ._create_creg(len(circ.qubits), "meas")
-        circ.add_register(new_creg)
-        circ.barrier()
-        circ.measure(circ.qubits, new_creg)
+        if add_creg:
+            new_creg = circ._create_creg(len(circ.qubits), "meas")
+            circ.add_register(new_creg)
+            circ.barrier()
+            circ.measure(circ.qubits, new_creg)
+        else:
+            if len(circ.clbits) < len(circ.qubits):
+                raise CircuitError(
+                    "The size of the ClassicalRegister must be equal or greater than"
+                    "the size of the QuantumRegister."
```

Comment:
It's more that the number of classical bits must be greater than or equal to the number of qubits - it's possible to have more than one (or even zero!) registers in the circuit.

(Also, there will be a missing space in the message between "than" and "the".)